### PR TITLE
Update release action to remove CI Build Wheel

### PR DESCRIPTION
**Pull request by MattToast**
Removes the use of CI Build Wheel now that SmartSim is a pure python package.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -32,108 +32,61 @@ on:
...
-        with:
-          path: ./wheelhouse/*.whl
-
-
-  build_sdist:
-    name: Build source distribution
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
+          sudo apt-get install -y build-essential wget
```
Comment (ashao) [inline .github/workflows/release.yml:52]: Do we actually even need this step?

**Review by ashao — APPROVED** [review]
Thanks for tracking this down. One suggestion to simplify even further, but we can always test this later

File: .github/workflows/release.yml (hunk 1 above)
Comment (MattToast) [reply .github/workflows/release.yml:52]: Almost certainly not, but I was afraid to remove more than I had to. I'll throw up a ticket to address after release

Comment (MattToast) [reply .github/workflows/release.yml:52]: Ticket posted [here](https://github.com/CrayLabs/SmartSim/issues/729)

**Pull request merged**